oak-authorization-principalbased/src/main/java/org/apache/jackrabbit/oak/spi/security/authorization/principalbased/impl/FilterProviderImpl.java

```diff
@@ -98,19 +107,23 @@ public Filter getFilter(@NotNull SecurityProvider securityProvider, @NotNull Roo
 
     //----------------------------------------------------< SCR Integration >---
 
+    public FilterProviderImpl() {
+        // constructor to use from SCR (not yet possible to use constructor injection, see https://issues.apache.org/jira/browse/OAK-9837)
+    }
+
     @Activate
-    protected void activate(Configuration configuration, Map<String, Object> properties) {
-        setPath(configuration);
+    protected void activate(Configuration configuration) {
+        setPath(configuration.path());
     }
 
     @Modified
-    protected void modified(Configuration configuration, Map<String, Object> properties) {
-        setPath(configuration);
+    protected void modified(Configuration configuration) {
+        setPath(configuration.path());
     }
 
-    private void setPath(@NotNull Configuration configuration) {
-        checkState(isValidPath(configuration.path()), "Configured path must be a valid absolute path.");
-        oakPath = configuration.path();
+    private void setPath(String path) {
```

Comment:
please add @Nullable or @NotNull annotation. was there a reason for removing the annotation?